src/css_parser.rs

```diff
@@ -2087,23 +2188,28 @@ fn parse_direction<'a>(input: &'a str)
     };
 
     if let Some(angle_type) = angle {
-        match angle_type {
+        let deg = match angle_type {
```

Comment:
If you are already defaulting to degree, then just remove the Some / None switching before this, i.e.:

```rust
 let angle = {
     if first_input.ends_with("grad") { AngleType::Gon }
     else if first_input.ends_with("rad") { AngleType::Rad }
     else if first_input.ends_with("deg") || first_input.is_numeric() /* note: not is_empty() */ { AngleType::Deg }
     else { /* return error, is something else than deg, grad or rad */  return Err(); }
};
```
This way you can avoid the `if let Some()` statement. Actually, you can skip the whole match statement and just parse direction in this if / else statement. Previously I didn't do that because I didn't want to default to degrees, but since it's necessary for the hsl to work correctly, it would be the way to go.

Reply:
Following your previous logic about having a missing `px` be an error for the sake of being explicit, I also made specifying `deg` a requirement for directions. Should I change that?

Reply:
Hmm... well, in order to do copy something like `hsl(180, 55%, 100%)`, which is the "standard" form on the web, you would need to default to deg. So I think that for consistency, you'd also need to default to px for units. While I don't like it, it's necessary for some things to work (so people can copy-paste styles from the web).

So yeah, just default to px and deg for both things. In the end it seems to be the better option.